File: palindrome.php
// A recursive function that  
// check a str[s..e] is  
// palindrome or not. 
function isPalRec($str, $s,$e) 
{ 
    // If there is only one character 
    if ($s == $e) 
    return true; 
  
    // If first and last  
    // characters do not match 
    if ($str[$s] != $str[$e]) 
    return false; 
  
    // If there are more than two  
    // characters, check if middle  
    // substring is also palindrome or not. 
    if ($s < $e + 1) 
    return isPalRec($str, $s + 1, $e - 1); 
  
    return true; 
} 
  
function isPalindrome($str) 
{ 
$n = strlen($str); 
  
// An empty string is  
// considered as palindrome 
if ($n == 0) 
    return true; 
  
return isPalRec($str, 0, $n - 1); 
} 
  
// Driver Code 
{ 
    $str = "geeg"; 
  
    if (isPalindrome($str)) 
    echo("Yes"); 
    else
    echo("No"); 
  
    return 0; 
} 
  
// This code is contributed  
// by nitin mittal. 
?> 

File: palindrome.py
# function to check string is 
# palindrome or not 
def isPalindrome(str): 

	# Run loop from 0 to len/2 
	for i in x range(0, len(str)/2): 
		if str[i] != str[len(str)-i-1]: 
			return False
	return True

# main function 
s = "malayalam"
ans = isPalindrome(s) 

if (ans): 
	print("Yes") 
else: 
	print("No") 